### DISCUSSION TITLE
Change the MQTT base topic for states, replacing "system_sensors"

### DISCUSSION THREAD

**Discussion by ingoratsdorf** [netalertx/NetAlertX#768] [Ideas]
At the moment, we use a state topic as follows (from the json config section):
`system-sensors/{sensor|binary_sensor|device_tracker}/{DEVICE_ID}/state`

Ideally, this should rather be
`{NETALERTX SELECTABLE BASE TOPIC}/{sensor|binary_sensor|device_tracker}/{DEVICE_ID}/state`
NETALERTX SELECTABLE BASE TOPIC would be `NetAlertX` by default, which would make it
`NetAlertX/{sensor|binary_sensor|device_tracker}/{DEVICE_ID}/state`

Basically, we'd replace the fixed `system_sensors` with a more meaningful value.

Reason being is that you'd see directly where the sensor belongs to.

For example, zigbee2mqtt uses
`zigbee2mqtt/{DEVICE_ID}/{various states]`

and openbeken uses
`openbeken/{DEVICE_ID}/{various states]`

Happy to develop, test and issue PR if that sounds like an acceptable solution.

**Comment by jokob-sk** [comment]
Hey,

Makes sense 🤔 it will however be a breaking change which I'd rather avoid if you have an idea on how. Ideally only new installs should default to the new root topic, so existing automations don't break.

**Comment by ingoratsdorf** [comment]
Hmm, the change would only break automations that directly call MQTT functions on the broker relating to the old topic.
With the changes proposed, the HA config topics would update and thus HA would automatically call the new topic for any given device or entity, thus your typical HA automations depending on device id or entity id would still continue to work.

I would not think of anyone calling MQTT functions directly, at least not in HA. There would be very little benefit or reason.

Outside of HA this would be different, keeping in mind that MQTT publish is not just for HA and others might be using it for functionality outside of HA.

It would be easy to allow for both old and new format in the script if we had an option to determine if its' an existing or a new install?

**Reply by jokob-sk** [reply]
Humm, what we could do is to release the first version with the value defaulting to "ssytem_sensors" and in the next update changing the default value to "NetAlertX". Yes, people might skip a version, but this should cover most installations. 

I was thinking, but I couldn't figure out an easy way how to determine if the plugin was loaded before. Only some convoluted workarounds came to mind, such as checking if there is already an entry in the Plugins_Objects table, or if there is a non-empty last_results file, or check the log entries when the MQTT plugin was executed. Maintaining such workarounds doesn't sound sustainable in the long term.

**Reply by ingoratsdorf** [reply]
Yeah, agree, leaving the default value for now to `system-sensors` makes most sense.
And I had similar 'workaround' thoughts and came to the same conclusion that inventing workarounds and complicated logic is typically not a good idea. Anything non-KISS has large potential to break.

Cool, I'll be working on something over the next couple'o days. 😄

**Reply by jokob-sk** [reply]
thanks a lot, @ingoratsdorf for the ongoing help and support 🙏looking forward to the PR. People like you help me keep my sanity maintaining NAX 😅
